lib/Elevate/Database.pm

```diff
@@ -0,0 +1,68 @@
+package Elevate::Database;
+
+=encoding utf-8
+
+=head1 NAME
+
+Elevate::Database
+
+Helper/Utility logic for database related tasks.
+
+=cut
+
+use cPstrict;
+
+use Elevate::OS ();
+
+use Cpanel::Pkgr ();
+
+use constant MYSQL_BIN => '/usr/sbin/mysqld';
+
+sub is_database_provided_by_cloudlinux ( $use_cache = 1 ) {
+
+    if ($use_cache) {
+        my $cloudlinux_database_installed = cpev::read_stage_file( 'cloudlinux_database_installed', '' );
+
+        # cloudlinux_database_installed should only ever be 1 or 0 if it is set
+        # by default, read_stage_file() will return '{}', but we are telling it to send back ''
+        # if cloudlinux_database_installed is not currently set
+        # This allows us to be sure that the cache is set when returning it
+        return $cloudlinux_database_installed if $cloudlinux_database_installed ne '';
```

Comment:
I suggest `length $cloudlinux_database_installed` instead to avoid `undef` warnings.

Reply:
Changed